test_coverage/api_testing.py

```diff
@@ -0,0 +1,157 @@
+from cgi import test
+import json
+import sys
+from fastapi.testclient import TestClient
+from app.api import API
+from data_generators.user_generators import Mentee, Mentor
+from pprint import pprint
+
+
+class TestingPlatform():
+    sys.tracebacklimit = 0
+    client = TestClient(API)
+    repopulate = False
+    def test_version(self):
+        """ Tests the current version of the API. """
+
+        response = self.client.get("/version")
+
+        assert response.status_code == 200, \
+            pprint("test_version: Expected Code 200. Got " +
+                   str(response.status_code))
+
+        assert response.json() == {'result': '0.43.7'}, \
+            pprint("test_version: Expected 0.43.7. Got " + str(response.json()))
+
+    def test_read(self, query=None):
+        """ Tests whether or not a collection can be read in"""
+
+        response = self.client.post(
+            "/Mentees/read", json=query or None)
+
+        assert response.status_code == 200, \
+            pprint("test_read: Expected Code 200. Got " +
+                   str(response.status_code))
+
+        assert response.json() != {}, \
+            pprint("test_read: Expected non empty json. Got " +
+                   str(response.json()))
+
+    def test_collections(self):
+        """ Tests collection name and size of each collection. """
+
+        response = self.client.get("/collections")
+
+        assert response.status_code == 200, \
+            pprint("test_collections: Expected Code 200. Got " +
+                   str(response.status_code))
+
+        assert response.json() != {}, \
+            "test_collections: Expected non empty JSON. Got " + \
+            str(response.json())
+
+    def test_update(self, query):
```

Comment:
query not used, and not passed when test_update called

Reply:
The test itself is failing at the assert. Not sure why, but I think we can make an exception and accept the pull request with the failing test.

Reply:
I've got it updated just need this to be done as a pr before I do another updating everything.